sdk/python/kfp/components/structures/components.json_schema.json

```diff
@@ -0,0 +1,389 @@
+
+    "GraphInputArgument": {
+      "description": "Represents the component argument value that comes from the graph component input.",
+      "type": "object",
+      "required": ["graphInput"],
+      "properties": {
+        "graphInput": {
+          "description": "References the input of the graph/pipeline.",
+          "type": "object",
+          "required": ["inputName"],
+          "properties": {
+            "inputName": {"type": "string"},
+            "type": {"$ref": "#/definitions/TypeSpecType"}
+          },
+          "additionalProperties": false
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "TaskOutputArgument": {
+      "description": "Represents the component argument value that comes from the output of a sibling task.",
+      "type": "object",
+      "required": ["taskOutput"],
+      "properties": {
+        "taskOutput": {
+          "description": "References the output of a sibling task.",
+          "type": "object",
+          "required": ["taskId", "outputName"],
+          "properties": {
+            "taskId": {"type": "string"},
+            "outputName": {"type": "string"},
+            "type": {"$ref": "#/definitions/TypeSpecType"}
+          },
+          "additionalProperties": false
+        }
+      },
+      "additionalProperties": false
+    },
+
+    "ArgumentType": {
+      "oneOf": [
+        {"type": "string"},
+        {"$ref": "#/definitions/GraphInputArgument"},
+        {"$ref": "#/definitions/TaskOutputArgument"}
+      ]
+    },
+
+    "TwoArgumentOperands": {
+      "description": "Pair of operands for a binary operation.",
+      "type": "object",
+      "required": ["op1", "op2"],
+      "properties": {
+        "op1":  {"$ref": "#/definitions/ArgumentType"},
+        "op2":  {"$ref": "#/definitions/ArgumentType"}
+      },
+      "additionalProperties": false
+    },
+
+    "TwoLogicalOperands": {
+      "description": "Pair of operands for a binary logical operation.",
+      "type": "object",
+      "required": ["op1", "op2"],
+      "properties": {
+        "op1":  {"$ref": "#/definitions/PredicateType"},
+        "op2":  {"$ref": "#/definitions/PredicateType"}
+      },
+      "additionalProperties": false
+    },
+
+    "PredicateType": {
+      "oneOf": [
+        {"type": "object", "required": ["=="], "properties": {"==":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+        {"type": "object", "required": ["!="], "properties": {"!=":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+        {"type": "object", "required": [">"], "properties": {">":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+        {"type": "object", "required": [">="], "properties": {">=":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+        {"type": "object", "required": ["<"], "properties": {"<":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+        {"type": "object", "required": ["<="], "properties": {"<=":  {"$ref": "#/definitions/TwoArgumentOperands"}}},
+
```

Comment:
nit: redundant blank line

Reply:
This was intentional - I wanted to separate comparison operators, logical operators and unary operators.
See the type of the operands: `TwoArgumentOperands`, `TwoLogicalOperands`, `PredicateType`.